v2/AdminCore.Services/EventService.cs

```diff
@@ -0,0 +1,178 @@
+                                                                              && x.Event.EmployeeId == employeeId);
+      return _mapper.Map<EventDto>(eventsBetweenDates);
+    }
+
+    public EventDto Get(int id)
+    {
+      var eventById = _databaseContext.EventRepository.Get(x => x.EventId == id);
+      return _mapper.Map<EventDto>(eventById);
+    }
+
+    public IList<EventDto> GetByStatusType(EventStatuses eventStatus, EventTypes eventType)
+    {
+      var events = _databaseContext.EventRepository.Get(x => x.EventStatus.EventStatusId == (int)eventStatus
+                                                             && x.EventType.EventTypeId == (int)eventType);
+      return _mapper.Map<List<EventDto>>(events);
+    }
+
+    public IList<EventDto> GetByType(EventTypes eventType)
+    {
+      var events = _databaseContext.EventRepository.Get(x => x.EventType.EventTypeId == (int)eventType);
+      return _mapper.Map<List<EventDto>>(events);
+    }
+
+    public DatabaseStatus SaveEvent(EventDto eventDto)
+    {
+      throw new NotImplementedException();
+    }
+
+    public DatabaseStatus UpdateEvent(EventDto eventDto)
+    {
+      var existingEvent = _databaseContext.EventRepository.GetById(eventDto);
+      if (existingEvent != null)
+      {
+        Event eventToUpdate = _mapper.Map(eventDto, existingEvent);
+
+        Save(eventToUpdate);
+        return DatabaseStatus.Success;
+      }
+
+      return DatabaseStatus.Failure;
+    }
+
+    public DatabaseStatus ApproveEvent(EventDto eventDto)
+    {
+      var existingEvent = _databaseContext.EventRepository.GetById(eventDto);
+      if (existingEvent != null)
+      {
+        Event eventToUpdate = existingEvent;
+        eventToUpdate.EventStatus.EventStatusId = (int)EventStatuses.Approved;
+        Save(eventToUpdate);
+        return DatabaseStatus.Success;
+      }
+
+      return DatabaseStatus.Failure;
+    }
+
+    public DatabaseStatus CancelEvent(EventDto eventDto)
+    {
+      var existingEvent = _databaseContext.EventRepository.GetById(eventDto);
+      if (existingEvent != null)
+      {
+        Event eventToCancel = existingEvent;
+        eventToCancel.EventStatus.EventStatusId = (int)EventStatuses.Cancelled;
+        _mapper.Map(eventDto, existingEvent);
+
+        Save(eventToCancel);
+        return DatabaseStatus.Success;
+      }
+
+      return DatabaseStatus.Failure;
+    }
+
+    public List<string> RejectEvent(int eventId, string message, int employeeId)
+    {
+      List<string> responses = new List<string>();
+      var existingEvent = _databaseContext.EventRepository.GetById(eventId);
+      if (existingEvent != null)
+      {
+        Event eventToReject = existingEvent;
+        eventToReject.EventStatus.EventStatusId = (int)EventStatuses.Rejected;
+        Employee employee = GetEmployeeFromEmployeeId(employeeId);
+        if (employee != null)
+        {
+          if (eventToReject.EventStatus.EventStatusId != (int)EventStatuses.Rejected)
```

Comment:
3 levels of nested ifs - make line 136-151 a method and look at extracting your nested ifs into further, smaller methods